### PR TITLE
util/ruby.sh: fix HOMEBREW_USE_RUBY_FROM_PATH to actually use user's PATH

### PR DESCRIPTION
**Pull request by Bo98**
Otherwise it just picks up from `/usr/bin/ruby` all the time.

### CODE REVIEW

File: Library/Homebrew/utils/ruby.sh [hunk 1]
```diff
@@ -13,30 +13,52 @@ test_ruby() {
...
+      do
+        rubies+=("${ruby_path}")
+      done < <(which -a ruby)
+    fi
+
+    # shellcheck disable=SC2230
+    while IFS= read -r ruby_path
+    do
+      rubies+=("${ruby_path}")
+    done < <(PATH="${HOMEBREW_PATH}" which -a ruby)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/ruby.sh:51]: Any way these two loops can get combined? Maybe a function?

File: Library/Homebrew/utils/ruby.sh [hunk 2]
```diff
@@ -13,30 +13,52 @@ test_ruby() {
...
+      # function which() is set by brew.sh
+      # it is aliased to `type -P`
+      # shellcheck disable=SC2230
+      while IFS= read -r ruby_path
+      do
+        rubies+=("${ruby_path}")
+      done < <(which -a ruby)
+    fi
+
+    # shellcheck disable=SC2230
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/ruby.sh:47]: Can this get a comment (or combined, like below)

**Review by MikeMcQuaid — APPROVED** [review]
Looks good! Some optional style suggestions.

File: Library/Homebrew/utils/ruby.sh (hunk 1 above)
Comment (Bo98) [reply Library/Homebrew/utils/ruby.sh:51]: Think I found a way

Comment (MikeMcQuaid) [reply Library/Homebrew/utils/ruby.sh]: thisistheway
👍🏻

**Comment by MikeMcQuaid** [comment]
Thanks again @Bo98!

**Pull request merged**